File: src/main/java/org/myfintech/payment/service/ClientService.java
package org.myfintech.payment.service;

import java.util.List;

import org.myfintech.payment.domain.ClientCreateDTO;
import org.myfintech.payment.domain.ClientDTO;
import org.myfintech.payment.entity.Client;

public interface ClientService {
    List<ClientDTO> findAll();
    ClientDTO findById(Long id);
    Client findEntityById(Long id);
    ClientDTO save(ClientCreateDTO dto);
    ClientDTO update(Long id, ClientDTO dto);
    void validate(ClientDTO dto);
}

File: src/main/kotlin/org/myfintech/payment/service/ClientService.kt
package org.myfintech.payment.service

import org.myfintech.payment.domain.ClientCreateDTO
import org.myfintech.payment.domain.ClientDTO
import org.myfintech.payment.entity.Client

interface ClientService {
    fun findAll(): List<ClientDTO>
    fun findById(id: Long): ClientDTO
    fun findEntityById(id: Long): Client
    fun save(dto: ClientCreateDTO): ClientDTO
    fun update(id: Long, dto: ClientDTO): ClientDTO
    fun validate(dto: ClientDTO)
}
